charts/replicated-library/templates/lib/_support_bundle_specs.tpl

```diff
@@ -0,0 +1,339 @@
+{{/*
+This template serves as a blueprint for all support bundle spec secret objects that are created
+within the replicated-library library.
+*/}}
+{{- define "replicated-library.support-bundle.spec" -}}
+  {{- $values := "" -}}
+  {{- if and (hasKey . "ContextValues") (hasKey .ContextValues "supportBundle") -}}
+    {{- $values = .ContextValues.supportBundle -}}
+  {{- else -}}
+    {{- fail "_support_bundle_specs.tpl requires the 'secret' ContextValues to be set" -}}
+  {{- end -}}
+  {{- $_ := set $.ContextValues "names" (dict "context" "supportBundle") -}}
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ include "replicated-library.names.prefix" . }}-support-bundle
+  labels:
+    troubleshoot.io/kind: support-bundle
+  data:
+    support-bundle-spec: 
+    {{- if .Values.supportBundle.installDefaultSpec -}}
+      {{ include "replicated-library.support-bundle.spec.default" . | b64enc | indent 1}}
+    {{- end }}
+{{- end }}
+
+{{- define "replicated-library.support-bundle.spec.default" -}}
```

Comment:
I haven't though this through yet, but I wonder if this should have defaults but be configurable? 
Or do we add configurable templates separate and you can just customize those directly?
I'm also curious if there's any way to keep us from having to sync this file with our troubleshoot repo, I don't have a solution to that off the top of my head just curious if you've thought about it.

Reply:
+1. I think we can have a default that uses the `URI` field inside of the chart. Perhaps that's all it includes and the user can also provide their own spec inside of the `values.yaml`. Would greatly condense the amount of code we include here as well.

Reply:
Thank you! Good idea, I have changed the default to use URI. 

For the customise one, I think this maybe a better way
```
supportBundle:
  default:
    enabled: true
    uri: https://raw.githubusercontent.com/replicatedhq/troubleshoot-specs/main/in-cluster/default.yaml
  custom:
    enabled: false
    uri: https://raw.githubusercontent.com/replicatedhq/troubleshoot-specs/main/in-cluster/some.yaml
    spec: {}
```

Reply:
For the current one, we need considerate the airgap situation, so the default spec has to be added. 
When use `helm template test . --debug `
you may see
```
stringData:
  support-bundle-spec:  "apiVersion: troubleshoot.sh/v1beta2\nkind: SupportBundle\nmetadata:\n  name: test-wg-easy-support-bundle-default\nspec:\n
    \ uri: https://raw.githubusercontent.com/replicatedhq/troubleshoot-specs/main/in-cluster/default.yaml\n
    \ collectors:\n    - clusterInfo: {}\n    - clusterResources: {}\n    - ceph: {}\n
``` 
However, after testing, it will not affect the secret, just way of helm display

Reply:
Tested that
if we remove
```
- exec:
        collectorName: kotsadm-rqlite-db
        name: kots/admin_console
        containerName: rqlite
        selector:
          - app=kotsadm-rqlite
        command:
          - sh
          - -c
          - |
            wget -qO- kotsadm:${RQLITE_PASSWORD}@localhost:4001/db/backup?fmt=sql  
        timeout: 10s
     
- clusterPodStatuses:
        outcomes:
          - fail:
              when: "!= Healthy"
              message: {{` "Status: {{ .Status.Reason }}" `}}
        
- clusterPodStatuses:
        checkName: contour pods unhealthy
        namespaces:
          - projectcontour
        outcomes:
          - fail:
              when: "!= Healthy" # Catch all unhealthy pods. A pod is considered healthy if it has a status of Completed, or Running and all of its containers are ready.
              message: {{` "A Contour pod, {{ .Name }}, is unhealthy with a status of {{ .Status.Reason }}. Restarting the pod may fix the issue." `}}
```
then `helm template test . --debug`
will have pretty yaml output

Reply:
A idea to sync this file with our troubleshoot repo
We can 
```
Data:
  support-bundle-spec: {{ .Files.Get spec.yaml | b64enc }}
```
we can create github action to monitor https://github.com/replicatedhq/kots/blob/main/pkg/supportbundle/defaultspec/spec.yaml and make a pr if the file has been changed. 

However, we have to use base64 encode to skip the issue that  we have `{{ .Name }}` and `{{ .Status.Reason }}` in our spec, which cause `toYaml` not working
```
outcomes:
          - fail:
              when: "!= Healthy" # Catch all unhealthy pods. A pod is considered healthy if it has a status of Completed, or Running and all of its containers are ready.
              message: A Contour pod, {{ .Name }}, is unhealthy with a status of {{ .Status.Reason }}. Restarting the pod may fix the issue. 
```